tests/credentials_tests.c

```diff
@@ -571,6 +579,7 @@ static int s_cached_credentials_provider_queued_async_test(struct aws_allocator
         s_verify_callback_status(&callback_results, 2, s_access_key_id_2, s_secret_access_key_2, s_session_token_2) ==
         0);
 
+    s_wait_for_provider_ref_cnt_to_become_one(mock_provider);
```

Comment:
I've got a different idea that gets at the root of the issue.

If the leak is due to the shutdown timing of the background thread used by the mock-async-provider, maybe we should use the `aws_global_thread_creator_shutdown_wait_for()` mechanism:
1) `aws_global_thread_creator_increment()` when the thread is created
2) `aws_global_thread_creator_decrement()` after the thread is joined and cleaned up
3) Tests using the mock-async-provider end with `aws_global_thread_creator_shutdown_wait_for()`

Reply:
This is not the issue. The issue is where the final refcount is released from the background thread so the join here fails